ocs_ci/ocs/resources/bucket_policy.py

```diff
@@ -78,21 +79,37 @@ def __init__(
         """
         self.account_name = name
         self.email_id = email
-        response = mcg.send_rpc_query(
-            api="account_api",
-            method="create_account",
-            params={
-                "email": email,
-                "name": name,
-                "has_login": admin_access,
-                "s3_access": s3_access,
-                "default_pool": backingstore_name,
-                "allowed_buckets": {
-                    "full_permission": full_bucket_access,
-                    "permission_list": buckets,
+        if float(config.ENV_DATA["ocs_version"]) < 4.9:
+            response = mcg.send_rpc_query(
+                api="account_api",
+                method="create_account",
+                params={
+                    "email": email,
+                    "name": name,
+                    "has_login": admin_access,
+                    "s3_access": s3_access,
+                    "default_pool": backingstore_name,
+                    "allowed_buckets": {
+                        "full_permission": full_bucket_access,
+                        "permission_list": buckets,
+                    },
                 },
-            },
-        ).json()
+            ).json()
+        else:
+            response = mcg.send_rpc_query(
+                api="account_api",
+                method="create_account",
+                params={
+                    "email": email,
+                    "name": name,
+                    "has_login": admin_access,
+                    "s3_access": s3_access,
+                    "allowed_buckets": {
+                        "full_permission": full_bucket_access,
+                        "permission_list": buckets,
+                    },
+                },
+            ).json()
```

Comment:
Should we instead create a new dict with all the params in it and remove the key 'default_pool' if the version is greater than 4.9 ?
something like,
params_dict =  "email": email,
                    "name": name,
                    "has_login": admin_access,
                    "s3_access": s3_access,
                    "default_pool": backingstore_name,
                    "allowed_buckets": {
                        "full_permission": full_bucket_access,
                        "permission_list": buckets,},}

 if version == 4.9 params_dict.pop('has_login') else params_dict
and then pass this params_dict as a value to arg:params
This way can reduce the code redundancy? wdyt?

Reply:
Ack. Made changes. Thanks :)